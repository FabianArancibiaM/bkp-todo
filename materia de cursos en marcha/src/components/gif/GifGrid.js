import React from 'react'
import { useFetchGifs } from '../../hooks/useFetchGifs';
import {GifGridItem} from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
    
    return (
        <div className="card-grid animate__animated animate__fadeIn">
            <h3>{category}</h3><br/>
                {loading && <p className="card-grid animate__animated animate__flash" >Loading...</p> }
                {
                    images.map( (img)=> 
                        <GifGridItem 
                        key={img.id}
                        { ...img } 
                        />
                    )
                }
        </div>
    )
}

