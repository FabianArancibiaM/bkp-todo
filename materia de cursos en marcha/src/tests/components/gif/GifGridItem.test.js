import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../../components/gif/GifGridItem';
describe('GifGridItem Test Funciones', () => {
    
    const title = 'Hola mundo';
    const url = 'https://localhost/algo.jpg';
    
    let wraper = shallow(<GifGridItem title={title} url={url} />);

    beforeEach( () => {
        wraper = shallow(<GifGridItem title={title} url={url} />);
    });

    test('GifGridItem', () => {
        expect(wraper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const textoParrafo = wraper.find('p').text();
        expect(textoParrafo.trim()).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wraper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('debe de tener animate__bounce ', () => {
        const className = wraper.find('div').prop('className');
        expect(className.includes('animate__bounce')).toBe(true);
    })
    
})