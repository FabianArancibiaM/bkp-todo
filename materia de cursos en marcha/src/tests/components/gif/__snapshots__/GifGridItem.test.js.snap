// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Test Funciones GifGridItem 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Hola mundo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Hola mundo
  </p>
</div>
`;
